kubespawner/reflector.py

```diff
@@ -140,20 +173,34 @@ def _watch_and_update(self):
                 if not self.first_load_future.done():
                     # signal that we've loaded our initial data
                     self.first_load_future.set_result(None)
+
+                # in case of timeout_seconds, the w.stream just exits (no exception thrown)
+                # -> we stop the watcher and start a new one
+
                 for ev in w.stream(
                         getattr(self.api, self.list_method_name),
-                        self.namespace,
-                        label_selector=self.label_selector,
-                        resource_version=resource_version,
+                        **watch_args
                 ):
                     cur_delay = 0.1
                     resource = ev['object']
+                    if not resource.metadata.resource_version:
+                        # reset resources_version if we don't have one
+                        del watch_args['resource_version']
+                    elif (watch_args.get('resource_version') != resource.metadata.resource_version):
+                        # only set resources_version if new resource_version is not None
+                        watch_args['resource_version'] = resource.metadata.resource_version
+                        self.log.info('Watch Event: {} - {}'.format(ev['type'], resource.metadata.name))
```

Comment:
this log probably shouldn't be sensitive to whether resource version is set. Maybe it's at the wrong indentation level?

Reply:
Only a trace / debug log. Just there to see that it actually happens. I'll cleanup and rebase the pull request.